lib/bitcoin/SidetreeTransactionParser.ts

```diff
@@ -0,0 +1,152 @@
+import BitcoinClient from './BitcoinClient';
+import BitcoinInputModel from './models/BitcoinInputModel';
+import BitcoinOutputModel from './models/BitcoinOutputModel';
+import BitcoinTransactionModel from './models/BitcoinTransactionModel';
+import SidetreeError from '../common/SidetreeError';
+import SidetreeTransactionModel from './models/SidetreeTransactionModel';
+
+/**
+ * Encapsulates functionality about parsing a sidetree transaction written on the bitcoin.
+ */
+export default class SidetreeTransactionParser {
+
+  public constructor (private bitcoinClient: BitcoinClient) {
+  }
+
+  /**
+   * Parses the given transaction and returns the sidetree transaction object.
+   *
+   * @param bitcoinTransaction The transaction to parse.
+   * @param sidetreePrefix The prefix of the sidetree transactions.
+   *
+   * @returns The data model if the specified transaction is a valid sidetree transaction; undefined otherwise.
+   */
+  public async parse (bitcoinTransaction: BitcoinTransactionModel, sidetreePrefix: string): Promise<SidetreeTransactionModel | undefined> {
+
+    const sidetreeData = this.getValidSidetreeDataFromOutputs(bitcoinTransaction.id, bitcoinTransaction.outputs, sidetreePrefix);
+
+    if (!sidetreeData) {
+      return undefined;
+    }
+
+    const writer = await this.getValidWriterFromInputs(bitcoinTransaction.id, bitcoinTransaction.inputs);
+
+    if (!writer) {
+      console.info(`Valid sidetree data was found but no valid writer was found for transaction id: ${bitcoinTransaction.id}`);
+      return undefined;
+    }
+
+    return {
+      data: sidetreeData,
+      writer: writer
+    };
+  }
+
+  private getValidSidetreeDataFromOutputs (transactionId: string, transactionOutputs: BitcoinOutputModel[], sidetreePrefix: string): string | undefined {

```

Comment:
It seems odd to pass `transactionId` just for logging purposes. I was thinking, maybe we tweak the requirement to accept the first output valid Sidetree data only, (and perhaps even only look at the first output only), then short circuit the code. This would help simplify the code and also reduce compute.

Reply:
Hmmm .. so the original behavior has been to actively remove any transactions which have multiple anchorstrings. I'm ok with your suggestion but the implementation does mean adding the outputIdx to the BitcoinOutputModel to ensure that we actually get the first one.

Reply:
Updating to recognize only the first anchorstring found in the outputs. The underline bitcore-lib returns the outputs in the order that they were written on the blockchain so we are going to take the order in which they were returned.